### DISCUSSION TITLE
Unsharpmask sometimes brightens the image?

### DISCUSSION THREAD

**Discussion by UndyingKrilin** [ImageMagick/ImageMagick#6881] [Help]
### ImageMagick version

7.0.8-Q16

### Operating system, version and so on

Windows 10 Enterprise, ver. 19045

### Description

I am having an issue with the ImageMagick's unsharpmask for some of my users. For most of them the image looks fine, but there is that one guy who gets a brighter image (repro 100%). After generating debug images after each step/command I do with imageMagick, I determined it was during unsharpmask step that the issue appeared.

I first suspected the image was not in a linear colorspace when executing the unsharpmask (which it was not), but after fixing that, it is still brighter than it should be (thought it did help a little).

I also had some feedback that a similar issue can appear randomly with a very low frequency for any user, but that could be an unrelated bug with the same visual effect.

It is difficult not to suspect a different setup that user could have, but wouldn't that affect more than just the unsharpmask? To get to the input images shown in this post, there is a successful colorFuzz, Trim, Extent, resize and then the unsharpmask. Only that last step causes the issue. I am using Magick++ but the corresponding command line would be:
```powershell
magick input.tga -fuzz 1.0 -trim -gravity center -background 'rgba(100%,100%,100%,0)' -extent 516x169 -colorspace RGB -resize 800x262 -unsharp 0x1+1+0.05 -colorspace sRGB out.tga
``` 

As seen above, the unsharpmask parameter I also use in Magick++ are: radius(0), sigma(1), amount(1), threshold(0.05).
The debug image are also generated by Magick++, after the initial copy, I convert it to png and annotate the gamma.

Here you can see after the resize and before the unsharpmask (2 above pistols) with RGB on the left and sRGB on the right. Then, the two on the bottoms are after the unsharpmask, again RGB on the left and sRGB on the right:
![Issue_steps](https://github.com/ImageMagick/ImageMagick/assets/26191843/c9e30b1a-aa58-4021-940a-6e599299e11e)

Finally, here is the same command's output I get on all the other users, or what I expect:
![Working_steps](https://github.com/ImageMagick/ImageMagick/assets/26191843/00d5a2e8-24cb-4d76-a6c2-009e5679bc98)

Any help would be greatly appreciated!

**Comment by snibgo** [comment]
Doing the unsharp *after* `-extent` seems strange.

Have you checked that the command line shows the same problem as your Magick++ program?

Some IM operations check whether the image is encoded as sRGB and, if it isn't, converts it to sRGB. I don't know if that is affecting you. If it is, then try changing ...
```
-colorspace RGB -resize 800x262 -unsharp 0x1+1+0.05 -colorspace sRGB
```
... to ...
```
-colorspace RGB -set colorspace sRGB -resize 800x262 -unsharp 0x1+1+0.05 -set colorspace RGB -colorspace sRGB
```
This tells IM that the colorspace is sRGB, so the resize and unsharp won't change the colorspace.

If that doesn't fix the problem, please post a sample input image. The actual image file, not a screenshot.

**Reply by UndyingKrilin** [reply]
Hi Snibgo, thank you for your answer!
I am doing the extent first to add empty border to my image after the trim, the unsharp is to remove the blurriness after the resize. Maybe that's not the way to do it, I am not too experimented in image processing... Note that the extent paremeters are determined in code and I just put them as value for simplicity (but for this image they will always be those you see in the initial post so it's safe to use as is for similar results to mine)

I did try locally the image magick command line, I get the same result as my Magick++ code. I did not try with the problematic user as he only gets the .xml and .dll from the user build. Furthermore, he's not a technical person and in another country/timezone, but if it's an important test to determine what is going on here, I will get it done.

I tried the suggested solution, thought it did not make much sense to me to trick IM into thinking the image is sRBG when it's not. In addition, I've read multiple time that IM assumes you know what you're doing and expect a linear colorspace for that sort of operations...
Anyway, same result unfortunately. Here are the proper images:

Here is the 6 debug images I generate from before the resize to after the unsharpmask, also the input .tga (which is not supported on github it seems, hence the .zip). The 6 images were generated with the suggested modification from your previous comment.
[input_images.zip](https://github.com/ImageMagick/ImageMagick/files/13450725/input_images.zip)

Sorry for the delay, with the timezone and all between me and the user, it takes sometime to get feedback on any proposed solutions.
Thank you in advance for your help!

**Reply by UndyingKrilin** [reply]
Hi Snibgo, any new ideas? Sorry to bother you, I just want to keep this thread active ...

**Comment by fmw42** [comment]
I suggest that you put a +write tmpX.png after each step and find out where the brightening occurs in your command

**Reply by UndyingKrilin** [reply]
I did, it occurs during the -unsharp step :)
The four pictures in my original post are in RGB and sRGB before the unsharp then after the unsharp.

**Comment by snibgo** [comment]
With IM v7.1.1-20 on Windows 11, I can't reproduce the problem.

With this command:
```
%IMG7%magick ^
  Input_Image(2000x1200).tga ^
  -auto-orient ^
  -trim +repage ^
  +write xc.png ^
-colorspace RGB ^
  -unsharp 0x1+1+0.05 ^
-colorspace sRGB ^
  uc.png
```
... the outputs `xc.png` and `uc.png` look the same lightness to me. Of course, some pixels in `uc.png` have become lighter and some have become darker.

If I remove the `-colorspace` options, the outputs again seem to be the same lightness.
